cadence/contracts/bridge/FlowEVMBridgeUtils.cdc

```diff
@@ -727,6 +781,198 @@ contract FlowEVMBridgeUtils {
+        let toPreBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        // Mint tokens to the recipient
+        let mintResult: EVM.Result = FlowEVMBridgeUtils.call(
+            signature: "mint(address,uint256)",
+            targetEVMAddress: erc20Address,
+            args: [to, amount],
+            gasLimit: 15000000,
+            value: 0.0
+        )
+        assert(mintResult.status == EVM.Status.successful, message: "Tranfer to bridge recipient failed")
+        // Ensure bridge to recipient was succcessful
+        let toPostBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        assert(
+            toPostBalance == toPreBalance + amount,
+            message: "Transfer to bridge recipient failed"
+        )
+    }
+
+    /// Transfers ERC20 tokens to the recipient and confirms that the recipient's balance was incremented and the escrow
+    /// balance was decremented by the requested amount.
+    ///
+    access(account)
+    fun mustTransferERC20(to: EVM.EVMAddress, amount: UInt256, erc20Address: EVM.EVMAddress) {
+        let bridgeCOAAddress = self.getBridgeCOAEVMAddress()
+
+        let toPreBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        let escrowPreBalance = FlowEVMBridgeUtils.balanceOf(
+            owner: bridgeCOAAddress,
+            evmContractAddress: erc20Address
+        )
+
+        // Mint tokens to the recipient
+        let mintResult: EVM.Result = FlowEVMBridgeUtils.call(
+            signature: "transfer(address,uint256)",
+            targetEVMAddress: erc20Address,
+            args: [to, amount],
+            gasLimit: 15000000,
+            value: 0.0
+        )
+        assert(mintResult.status == EVM.Status.successful, message: "Tranfer to bridge recipient failed")
+
+        // Ensure bridge to recipient was succcessful
+        let toPostBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        let escrowPostBalance = FlowEVMBridgeUtils.balanceOf(
+            owner: bridgeCOAAddress,
+            evmContractAddress: erc20Address
+        )
+        assert(
+            toPostBalance == toPreBalance + amount,
+            message: "Transfer to bridge COA failed - cannot bridge FT without bridge escrow"
+        )
+        assert(
+            escrowPostBalance == escrowPreBalance - amount,
+            message: "Transfer to bridge COA failed - cannot bridge FT without bridge escrow"
+        )
+    }
+
+    /// Executes the provided method, assumed to be a protected transfer call, and confirms that the transfer was
+    /// successful by validating that the named owner's balance was decremented by the requested amount and the bridge
+    /// escrow balance was incremented by the same amount.
+    ///
+    access(account)
+    fun mustExecuteProtectedERC20TransferCall(
```

Comment:
Maybe I am not understanding what a protected erc20 transfer is, but shouldn't this function name indicate that the tokens are being transferred to the bridge instead of just a regular transfer?

Reply:
Yeah good point, maybe something like `mustEscrowERC20`?